### ISSUE TITLE
How to add any other pod instead of OneSignal

### ISSUE THREAD

**Issue by dhruvpvx** [nikwebr/expo-notification-service-extension-plugin#2]
I'm trying to integrate https://docs.wonderpush.com/docs/adding-a-notification-service-extension with expo. but in the plugin there is no mention of adding any other pod instead of OneSignal. Please help me if anyone else had any luck integrating that

**Comment by nikwebr** [comment]
Indeed, the plugin does not support installing other pods.  You could give this tutorial a try: [https://docs.wonderpush.com/docs/mobile-push-notifications-react-native#step-3-add-the-sdk](https://docs.wonderpush.com/docs/mobile-push-notifications-react-native#step-3-add-the-sdk)
Install the react native dependencies, and instead of ejecting your expo app use this plugin to setup the Notification Service Extension. And follow the expo guides for enabling the required capabilities.

Wish you the best and if it works, please let me know :)

**Comment by OneUne** [comment]
I solved this by creating a custom Config Plugin that works alongside expo-notification-service-extension-plugin:

  Step 1: Create plugins/withNotificationServiceExtension.js

```
  const { withPodfile } = require('expo/config-plugins');

  function withNotificationServiceExtension(config, { pods = [], iosDeploymentTarget } = {}) {
    return withPodfile(config, (config) => {
      const podfileContent = config.modResults.contents;

      if (!podfileContent.includes("target 'NotificationServiceExtension'")) {
        const nseTarget = `
  target 'NotificationServiceExtension' do
...
  ${pods.map((pod) => `  pod '${pod}'`).join('\n')}
  end
  `;

        config.modResults.contents += nseTarget;
        console.log(`✅ NotificationServiceExtension target added to Podfile`);
      } else {
        console.log('ℹ️  NotificationServiceExtension target already exists in Podfile');
      }

      return config;
    });
  }

  module.exports = withNotificationServiceExtension;
```

  Step 2: Add both plugins to app.config.js

```
  plugins: [
    // ... other plugins
    [
      "expo-notification-service-extension-plugin",
      {
        mode: "development",
        iosNSEFilePath: "notification-service-extension/NotificationService.m"
      }
    ],
    ["./plugins/withNotificationServiceExtension", {
      pods: [], // Add any pods you need
    }],
  ]
```

  Step 3: Run prebuild
```
  npx expo prebuild --clean
```

  This approach:
  - ✅ Uses expo-notification-service-extension-plugin for basic NSE setup
  - ✅ Uses withPodfile for Podfile modifications
  - ✅ Automatically adds pods to NotificationServiceExtension target
  - ✅ Works with any CocoaPods dependencies (Firebase, OneSignal, etc.)
  - ✅ No manual Podfile editing required

  The two plugins work together perfectly - the first creates the NSE structure, the second adds the
  necessary dependencies using Expo's official Podfile modification API.
